### ISSUE TITLE
visudo is installed as part of sudo bundle but immediately gives error that  vi editor is not found

### ISSUE THREAD

**Issue by eadamsintel** [clearlinux/clr-bundles#105]
Either nano needs to be made the default editor or vi should be installed in the default configuration.

**Comment by fenrus75** [comment]
what is your EDITOR env variable set to ?

**Comment by eadamsintel** [comment]
eadams@otc-tme-wf-02 ~ $ echo $EDITOR
/usr/bin/nano
eadams@otc-tme-wf-02 ~ $ sudo visudo
visudo: no editor found (editor path = /usr/bin/vi)

This works   
eadams@otc-tme-wf-02 ~ $ sudo EDITOR=nano visudo

**Comment by fenrus75** [comment]
the sudo command erases the environment

don't do that


sudo su -
visudo

will work

**Comment by eadamsintel** [comment]
Good point. I had not considered that. However, whenever you search online for sudoless passwords or guides on visudo you always see it as  "sudo visudo"   On other OS installs like CentOS "sudo visudo" works without having to sudo su - first.  This used to work on previous installs of Clear Linux from months ago as I just tried it.  

BTW On my install of Clear linux with the newer installer where I setup an account and gave it admin privileged  I get errors when I run that command.  I believe this was noted in another issue somewhere else but I can't find it. 

$ sudo su -
You are required to change your password immediately (administrator enforced)
su: Authentication token manipulation error

sudo -s works though to change to root and then #visudo works

**Comment by fenrus75** [comment]
just to get sudo access you must not use visudo

you just add the user to "wheel" or "wheelnopw" group.

visudo is only for if you want to have custom, more complex, rules.

**Comment by eadamsintel** [comment]
Yes. I was trying to setup no password sudo for one of my systems and needed visudo for that. If I edit the sudoers file directly I mess it up and get locked out of my system.  Going back to my original reason for posting this is that every guide online I saw for setting up a no password sudo says to do "sudo visudo" and that didn't just work. It seems like it should.

**Comment by fenrus75** [comment]
to set up no-password sudo you do

usemod -aG wheelnopw  <username>

you do NOT use visudo. ever. for this use case

**Comment by fenrus75** [comment]
(usermod, not usemod)

**Comment by eadamsintel** [comment]
Thanks and good to know. If you search "no password sudo" on ddg the first 7 links I checked all say to use "sudo visudo" and the first link is an Ubuntu one.  I always thought that was the way to do it and is what is being recommended to others.   I think others would be confused as well but if someone is bold enough to mess around with visudo then they would likely know how to set the Editors variable or switch to root.
